core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -89,52 +92,57 @@ public int run(final String[] args, final Properties config) {
 
         int exitCode = EXIT_CODE_SUCCESS;
 
-        AdminClient adminClient = null;
-        ZkUtils zkUtils = null;
+        KafkaAdminClient kafkaAdminClient = null;
+
         try {
             parseArguments(args);
             dryRun = options.has(dryRunOption);
 
-            adminClient = AdminClient.createSimplePlaintext(options.valueOf(bootstrapServerOption));
             final String groupId = options.valueOf(applicationIdOption);
 
+            validateNoActiveConsumers(groupId);
 
-            zkUtils = ZkUtils.apply(options.valueOf(zookeeperOption),
-                30000,
-                30000,
-                JaasUtils.isZkSecurityEnabled());
+            Properties adminClientProperties = new Properties();
+            adminClientProperties.put("bootstrap.servers", options.valueOf(bootstrapServerOption));
+            kafkaAdminClient = (KafkaAdminClient) AdminClient.create(adminClientProperties);
 
             allTopics.clear();
-            allTopics.addAll(scala.collection.JavaConversions.seqAsJavaList(zkUtils.getAllTopics()));
-
-
-            if (!adminClient.describeConsumerGroup(groupId, 0).consumers().get().isEmpty()) {
-                throw new IllegalStateException("Consumer group '" + groupId + "' is still active. " +
-                            "Make sure to stop all running application instances before running the reset tool.");
-            }
+            allTopics.addAll(kafkaAdminClient.listTopics().names().get(60, TimeUnit.SECONDS));
 
             if (dryRun) {
                 System.out.println("----Dry run displays the actions which will be performed when running Streams Reset Tool----");
             }
             maybeResetInputAndSeekToEndIntermediateTopicOffsets();
-            maybeDeleteInternalTopics(zkUtils);
+            maybeDeleteInternalTopics(kafkaAdminClient);
 
         } catch (final Throwable e) {
             exitCode = EXIT_CODE_ERROR;
             System.err.println("ERROR: " + e);
             e.printStackTrace(System.err);
         } finally {
-            if (adminClient != null) {
-                adminClient.close();
-            }
-            if (zkUtils != null) {
-                zkUtils.close();
+            if (kafkaAdminClient != null) {
+                kafkaAdminClient.close();
             }
         }
 
         return exitCode;
     }
 
+    private void validateNoActiveConsumers(String groupId) {
+        kafka.admin.AdminClient olderAdminClient = null;
+        try {
+            olderAdminClient = kafka.admin.AdminClient.createSimplePlaintext(options.valueOf(bootstrapServerOption));
```

Comment:
Could we add a TODO marker to remove this when `describeGroup` is added to the admin client in case we forget about it?

Reply:
We should create a JIRA instead of adding a TODO...

Reply:
added JIRA https://issues.apache.org/jira/browse/KAFKA-5965